### DISCUSSION TITLE
Separate logs for separate configurations

### DISCUSSION THREAD

**Discussion by haydenbaker** [jreleaser/jreleaser#1630] [Ideas]
We are using the gradle plugin in our [project](https://github.com/smithy-lang/smithy). We have a repository where we have a [JReleaser configuration](https://github.com/smithy-lang/smithy/blob/main/build.gradle#L294) in the root (for release jars), and a separate [JReleaser configuration](https://github.com/smithy-lang/smithy/blob/main/smithy-cli/build.gradle#L263) within one of the subprojects (for releasing a CLI). 

If we decide to run a full release for both "projects" concurrently, it would be nice to have the `trace.log` file be generated in each project's respective `build/` folder. Currently (at least with configs), only a single log file gets generated at the root build-dir, and the logs are interlaced, which makes it difficult to read through.

**Comment by aalmiray** [comment]
I think the problem stems from this piece of code

https://github.com/jreleaser/jreleaser/blob/3ccb2e4fcb0ab9510f8171eae1e20893d7dd0f97/plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/JReleaserProjectConfigurer.groovy#L68-L88

The logger is set as a Gradle BuildService which treats it as a singleton for the whole build. Given that the root project is initialized before the children results in the logger pointing to the root's build and reused by the children. Some releases ago we switched to using a build service because:

1. Windows messed up cleaning operations (as always due to still open file descriptors and file permissions).
2. Build services are the recommended idiom since .. Gradle 7 I think?
3. Typically JReleaser should be configured on a single project.

We'd have to re-evaluate how to handle multiple configurations on a multi-project build as the current implementation won't let you do what you want i.e, invoking the plugin in multiple projects as part of the same build session.

**Reply by haydenbaker** [reply]
Good information to know. I knew there would be weirdness with running multiple configs in a mono-repo. We'll be executing a release soon with the multi-config setup, but I didn't seem to have any issues when testing (outside of log readability). I'll report back on real results afterwards.
